packages/app-builder/src/components/Scenario/AstBuilder/AstBuilder.tsx

```diff
@@ -45,14 +46,16 @@ export function AstBuilder({
       <CopyPasteASTContextProvider>
         <TimeAddEditModal>
           <AggregationEditModal>
-            <RootAstBuilderNode
-              setOperand={setOperand}
-              setOperator={setOperator}
-              appendChild={appendChild}
-              remove={remove}
-              editorNodeViewModel={editorNodeViewModel}
-              viewOnly={viewOnly}
-            />
+            <FuzzyMatchComparatorEditModal>
```

Comment:
This is anoying and non scallable.

The current pattern is : 
- use a single modal for each specific edit
- share a context
- use it when required

With that we have issues when :
- nesting modals: we only share a single modal comp for each specific node type (meaning we can't nest them)
- composing modals: depending on which modal need which other modal, the order of composition declared in this file matter.


The long term better plan would be to create a modal locally, for each specific place we need one.